src/main/java/org/folio/rest/impl/TenantSampleApi.java

```diff
@@ -0,0 +1,124 @@
+        return;
+      }
+
+      if (isLoadSample(tenantAttributes)) {
+        log.info("Loading sample data...");
+        String tenantId = TenantTool.calculateTenantId(headers.get(RestVerticle.OKAPI_HEADER_TENANT));
+
+        allOf(sampleDataMap.entrySet().stream()
+          .map(e -> loadSample(e.getKey(), getMockAsJson(SAMPLES_PATH + "/" + e.getValue()).mapTo(LogRecord.class), context, tenantId))
+          .toArray(CompletableFuture[]::new))
+          .thenAccept(vVoid -> handlers.handle(io.vertx.core.Future.succeededFuture(PostTenantResponse
+                    .respond201WithApplicationJson(""))))
+          .exceptionally(throwable -> {
+            handlers.handle(io.vertx.core.Future.succeededFuture(PostTenantResponse
+              .respond500WithTextPlain(throwable.getLocalizedMessage())));
+            return null;
+          });
+      } else {
+        handlers.handle(res);
+      }
+    }, context);
+  }
+
+  private CompletableFuture<Void> loadSample(String tableName, LogRecord logRecord, Context context, String tenantId) {
+    CompletableFuture<Void> future = new CompletableFuture<>();
+    PostgresClient.getInstance(context.owner(), tenantId).save(tableName, logRecord, reply -> {
+      if (reply.succeeded()) {
+        future.complete(null);
+      } else {
+        future.completeExceptionally(reply.cause());
+      }
+    });
+    return future;
+  }
+
+  @Override
+  public void deleteTenant(Map<String, String> headers, Handler<AsyncResult<Response>> handlers, Context context) {
+    super.deleteTenant(headers, res -> {
+      Vertx vertx = context.owner();
+      String tenantId = TenantTool.tenantId(headers);
+      PostgresClient.getInstance(vertx, tenantId)
+        .closeClient(event -> handlers.handle(res));
+    }, context);
+  }
+
+  private boolean isLoadSample(TenantAttributes tenantAttributes) {
+    // if a system parameter is passed from command line, ex: loadSample=true
+    // that value is considered,Priority of Parameters:
+    // Tenant Attributes > command line parameter > default(false)
+    boolean loadSample = Boolean.parseBoolean(MODULE_SPECIFIC_ARGS.getOrDefault(PARAMETER_LOAD_SAMPLE,
+      "false"));
+    List<Parameter> parameters = tenantAttributes.getParameters();
+    for (Parameter parameter : parameters) {
+      if (PARAMETER_LOAD_SAMPLE.equals(parameter.getKey())) {
+        loadSample = Boolean.parseBoolean(parameter.getValue());
+      }
+    }
+    return loadSample;
+  }
+
+  private JsonObject getMockAsJson(String fullPath) {
+    log.info("Using mock datafile: " + fullPath);
+    try (InputStream resourceAsStream = TenantSampleApi.class.getClassLoader().getResourceAsStream(fullPath)) {
+      if (resourceAsStream != null) {
+        return new JsonObject(IOUtils.toString(resourceAsStream, StandardCharsets.UTF_8));
+      }
+    } catch (IOException e) {
+      log.error("Failed to load mock data: " + fullPath, e);
+    }
+    return new JsonObject();
```

Comment:
From my perspective throwing an exception here is more appropriate than returning of empty json. If smth goes wrong with samples Tenant Api should return an error instead of population of empty data.